third_party/move/move-compiler-v2/tests/lambda/non_lambda_arg.lambda.exp

```diff
@@ -0,0 +1,416 @@
+module 0x42::sort {
+    use std::vector;
+    public fun incorrect_sort<T>(arr: &mut vector<#0>,a_less_b: |(#0, #0)|bool) {
+        {
+          let n: u64 = vector::length<T>(Freeze(false)(arr));
+          sort::incorrect_sort_recursive<T>(arr, 0, Sub<u64>(n, 1), a_less_b)
+        }
+    }
+    public fun incorrect_sort_recursive<T>(arr: &mut vector<#0>,low: u64,high: u64,a_less_b: |(#0, #0)|bool) {
+        if Lt<u64>(low, high) {
+          {
+            let pi: u64 = Add<u64>(low, Div<u64>(high, 2));
+            sort::incorrect_sort_recursive<T>(arr, low, Sub<u64>(pi, 1), a_less_b);
+            sort::incorrect_sort_recursive<T>(arr, Add<u64>(pi, 1), high, a_less_b);
+            Tuple()
+          }
+        } else {
+          Tuple()
+        };
+        Tuple()
+    }
+} // end 0x42::sort
+
+
+// -- Model dump after env processor specification rewriter:
+module 0x42::sort {
+    use std::vector;
+    public fun incorrect_sort<T>(arr: &mut vector<#0>,a_less_b: |(#0, #0)|bool) {
+        {
+          let n: u64 = vector::length<T>(Freeze(false)(arr));
+          sort::incorrect_sort_recursive<T>(arr, 0, Sub<u64>(n, 1), a_less_b)
+        }
+    }
+    public fun incorrect_sort_recursive<T>(arr: &mut vector<#0>,low: u64,high: u64,a_less_b: |(#0, #0)|bool) {
+        if Lt<u64>(low, high) {
+          {
+            let pi: u64 = Add<u64>(low, Div<u64>(high, 2));
+            sort::incorrect_sort_recursive<T>(arr, low, Sub<u64>(pi, 1), a_less_b);
+            sort::incorrect_sort_recursive<T>(arr, Add<u64>(pi, 1), high, a_less_b);
+            Tuple()
+          }
+        } else {
+          Tuple()
+        };
+        Tuple()
+    }
+} // end 0x42::sort
+
+
+============ initial bytecode ================
+
+[variant baseline]
+public fun sort::incorrect_sort<#0>($t0: &mut vector<#0>, $t1: |(#0, #0)|bool) {
+     var $t2: u64
+     var $t3: &vector<#0>
+     var $t4: u64
+     var $t5: u64
+     var $t6: u64
+  0: $t3 := freeze_ref(implicit)($t0)
+  1: $t2 := vector::length<#0>($t3)
+  2: $t4 := 0
+  3: $t6 := 1
+  4: $t5 := -($t2, $t6)
+  5: sort::incorrect_sort_recursive<#0>($t0, $t4, $t5, $t1)
+  6: return ()
+}
+
+
+[variant baseline]
+public fun sort::incorrect_sort_recursive<#0>($t0: &mut vector<#0>, $t1: u64, $t2: u64, $t3: |(#0, #0)|bool) {
+     var $t4: bool
+     var $t5: u64
+     var $t6: u64
+     var $t7: u64
+     var $t8: u64
+     var $t9: u64
+     var $t10: u64
+     var $t11: u64
+  0: $t4 := <($t1, $t2)
+  1: if ($t4) goto 2 else goto 13
+  2: label L0
+  3: $t7 := 2
+  4: $t6 := /($t2, $t7)
+  5: $t5 := +($t1, $t6)
+  6: $t9 := 1
+  7: $t8 := -($t5, $t9)
+  8: sort::incorrect_sort_recursive<#0>($t0, $t1, $t8, $t3)
+  9: $t11 := 1
+ 10: $t10 := +($t5, $t11)
+ 11: sort::incorrect_sort_recursive<#0>($t0, $t10, $t2, $t3)
+ 12: goto 14
+ 13: label L1
+ 14: label L2
+ 15: return ()
+}
+
+
+Diagnostics:
+error: local `a_less_b` of type `|(T, T)|bool` does not have the `copy` ability
```

Comment:
I am assuming we should we ignore weird errors in `lambda.exp` for now, because these will change once lambdas are implemented?